File: backend/NoteAppAPI/src/main/kotlin/org/example/noteappapi/config/FirebaseConfig.java
package org.example.noteappapi.config;

// FirebaseConfig.kt

import com.google.auth.oauth2.GoogleCredentials;
import com.google.firebase.FirebaseApp;
import com.google.firebase.FirebaseOptions;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.cloud.FirestoreClient;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import java.io.FileInputStream;
import javax.annotation.PostConstruct;

@Configuration
class FirebaseConfig {

    @PostConstruct
    fun initialize() {
        try {
            // Last ned serviceAccountKey.json fra Firebase Console
            val serviceAccount = FileInputStream("path/to/serviceAccountKey.json")

            val options = FirebaseOptions.builder()
                    .setCredentials(GoogleCredentials.fromStream(serviceAccount))
                    .build()

            if (FirebaseApp.getApps().isEmpty()) {
                FirebaseApp.initializeApp(options);
            }
        } catch (e: Exception) {
            e.printStackTrace();
        }
    }

    @Bean
    fun firebaseAuth(): FirebaseAuth = FirebaseAuth.getInstance()

    @Bean
    fun firestore() = FirestoreClient.getFirestore()
}

File: backend/NoteAppAPI/src/main/kotlin/org/example/noteappapi/config/FirebaseConfig.kt
package org.example.noteappapi.config

import com.google.auth.oauth2.GoogleCredentials
import com.google.cloud.firestore.Firestore
import com.google.firebase.FirebaseApp
import com.google.firebase.FirebaseOptions
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.cloud.FirestoreClient
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.io.FileInputStream

@Configuration
class FirebaseConfig {

    @Bean
    fun firebaseApp(): FirebaseApp {
        val serviceAccountPath = System.getenv("FIREBASE_SERVICE_ACCOUNT_KEY") ?: "firebase-key.json"

        val credentials = GoogleCredentials.fromStream(
            FileInputStream(serviceAccountPath)
        )

        val options = FirebaseOptions.builder()
            .setCredentials(credentials)
            .build()

        return if (FirebaseApp.getApps().isEmpty()) {
            FirebaseApp.initializeApp(options)
        } else {
            FirebaseApp.getInstance()
        }
    }

    @Bean
    fun firebaseAuth(firebaseApp: FirebaseApp): FirebaseAuth {
        return FirebaseAuth.getInstance(firebaseApp)
    }

    @Bean
    fun firestore(firebaseApp: FirebaseApp): Firestore {
        return FirestoreClient.getFirestore(firebaseApp)
    }
}
